src/server/worker/pipeline/service/service.go

```diff
@@ -0,0 +1,118 @@
+package service
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/client/pps"
+	"github.com/pachyderm/pachyderm/src/server/pkg/ppsconsts"
+	"github.com/pachyderm/pachyderm/src/server/pkg/ppsutil"
+	"github.com/pachyderm/pachyderm/src/server/worker/common"
+	"github.com/pachyderm/pachyderm/src/server/worker/datum"
+	"github.com/pachyderm/pachyderm/src/server/worker/driver"
+	"github.com/pachyderm/pachyderm/src/server/worker/logs"
+	"github.com/pachyderm/pachyderm/src/server/worker/pipeline"
+)
+
+// Runs the given callback with the latest commit for the pipeline.  The given
+// context will be canceled if a newer commit is ready.
```

Comment:
Could you clarify in this comment that it calls the function multiple times as new commits come in? It's obvious from implementation but this comment makes it seem like it might only get called once.